From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Elliott <pelliott@ualberta.ca>
Date: Wed, 18 May 2022 01:43:41 -0600
Subject: [PATCH] Remove contents of return_unused_stack_to_os

guile attempts to madvise(2) away parts of the stack, but serenity only
supports madvise(2) on entire mmaped regions.
---
 libguile/vm.c | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/libguile/vm.c b/libguile/vm.c
index 6fd5c55..2eab110 100644
--- a/libguile/vm.c
+++ b/libguile/vm.c
@@ -626,30 +626,6 @@ scm_i_vm_prepare_stack (struct scm_vm *vp)
 static void
 return_unused_stack_to_os (struct scm_vm *vp)
 {
-#if HAVE_SYS_MMAN_H
-  uintptr_t lo = (uintptr_t) vp->stack_bottom;
-  uintptr_t hi = (uintptr_t) vp->sp;
-
-  lo &= ~(page_size - 1U); /* round down */
-  hi &= ~(page_size - 1U); /* round down */
-
-  /* Return these pages to the OS.  The next time they are paged in,
-     they will be zeroed.  */
-  if (lo < hi)
-    {
-      int ret = 0;
-
-      do
-        ret = madvise ((void *) lo, hi - lo, MADV_DONTNEED);
-      while (ret && errno == EAGAIN);
-
-      /* If the OS doesn't implement 'madvise' (as is currently the case
-         for GNU/Hurd), don't warn the user since there's nothing they
-         can do about it.  */
-      if (ret && errno != ENOSYS)
-        perror ("madvise failed");
-    }
-#endif
 }
 
 #define SLOT_MAP_CACHE_SIZE 32U
